ClassTranscribeServer/Controllers/CaptionsController.cs

```diff
@@ -141,26 +141,32 @@ public async Task<ActionResult<Caption>> CancelDownVote(string id)
 
         // POST: api/Captions
         [HttpGet("SearchInOffering")]
-        public async Task<ActionResult<IEnumerable<SearchedCaptionDTO>>> SearchInOffering(string offeringId, string query)
-        {
-
-
+        public async Task<ActionResult<IEnumerable<SearchedCaptionDTO>>> SearchInOffering(string offeringId, string query, string filterLanguage="")
+        { 
             var allVideos = await _context.Medias.Where(m => m.Playlist.OfferingId == offeringId)
                 .Select(m => new { m.VideoId, m.Video, MediaId = m.Id, m.PlaylistId, PlaylistName = m.Playlist.Name, MediaName = m.Name }).ToListAsync();
 
-            var captions = await _context.Medias.Where(m => m.Playlist.OfferingId == offeringId)
+            
+            var allOfferingCaptions = _context.Medias.Where(m => m.Playlist.OfferingId == offeringId)
                 .Select(m => m.Video).SelectMany(v => v.Transcriptions)
-                    .SelectMany(t => t.Captions)
-                    .Where(c => EF.Functions.ToTsVector("english", c.Text).Matches(query))
-                    .Take(100).Select(c => new SearchedCaptionDTO
+                    .Where(t => (string.IsNullOrWhiteSpace(filterLanguage) || t.Language == filterLanguage))
+                    .SelectMany(t => t.Captions);
+
+// ToTsVector is not implemented for the in-memory database used for testing
+
+            var matchingCaptions = _context.Database.IsNpgsql() ?
+                    allOfferingCaptions.Where(c => EF.Functions.ToTsVector("english", c.Text).Matches(query))
```

Comment:
question, doesn't this always search for english?

Reply:
No. the data can include languages. Tsvector is a postgres function. It builds a frequency map of words but it uses assumptions about English word frequency to ignore the most common words and to reduce words to a lexeme simplified form (e.g. reduce plurals to singuar). See-
https://www.postgresql.org/docs/9.1/textsearch-controls.html

This the above code will be suboptimal for non-English searches. we don't care too much because we will be switching over ElasticSearch searches anyway.